internal/buildsign/pod/manager.go

```diff
@@ -0,0 +1,133 @@
+package pod
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	kmmv1beta1 "github.com/kubernetes-sigs/kernel-module-management/api/v1beta1"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/api"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/buildsign"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/kernel"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/utils"
+)
+
+type podManager struct {
+	client              client.Client
+	maker               Maker
+	signer              Signer
+	buildSignPodManager BuildSignPodManager
+}
+
+func NewManager(client client.Client, maker Maker, signer Signer, buildSignPodManager BuildSignPodManager) buildsign.Manager {
+	return &podManager{
+		client:              client,
+		maker:               maker,
+		signer:              signer,
+		buildSignPodManager: buildSignPodManager,
+	}
+}
+
+func (pm *podManager) GetStatus(ctx context.Context, name, namespace, kernelVersion string,
+	action kmmv1beta1.BuildOrSignAction, owner metav1.Object) (kmmv1beta1.BuildOrSignStatus, error) {
+	podType := PodTypeBuild
+	if action == kmmv1beta1.SignImage {
+		podType = PodTypeSign
+	}
+	normalizedKernel := kernel.NormalizeVersion(kernelVersion)
+	foundPod, err := pm.buildSignPodManager.GetModulePodByKernel(ctx, name, namespace, normalizedKernel, podType, owner)
+	if err != nil {
+		if !errors.Is(err, ErrNoMatchingPod) {
+			return kmmv1beta1.BuildOrSignStatus(""), fmt.Errorf("failed to get pod %s/%s, action %s: %v", namespace, name, action, err)
+		}
+		return kmmv1beta1.BuildOrSignStatus(""), nil
+	}
+	status, err := pm.buildSignPodManager.GetPodStatus(foundPod)
+	if err != nil {
+		return kmmv1beta1.BuildOrSignStatus(""), fmt.Errorf("failed to get status from the pod %s/%s, action %s: %v",
+			foundPod.Namespace, foundPod.Name, action, err)
+	}
+	switch status {
+	case StatusCompleted:
+		return kmmv1beta1.ActionSuccess, nil
+	case StatusFailed:
+		return kmmv1beta1.ActionFailure, nil
+	}
+	return kmmv1beta1.BuildOrSignStatus(""), nil
```

Comment:
Since we should never reach this line, shouldn't we remove an error instead of `nil`?

Reply:
we will reach this line in case the Pod is still running or preparing to be run (pending). The error handling is done on line 51

Reply:
How about we add a comment for clarity?